inference-engine/src/inference_engine/ie_blob.cpp

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+
+#include "ie_blob.h"
+
+namespace InferenceEngine {
+
+Blob::Blob(const Blob& blob): tensorDesc(blob.tensorDesc), roiPtr(blob.roiPtr ? new ROIData(*blob.roiPtr) : nullptr) {}
+
+void Blob::setROI(const ROIData& roiData) {
+    roiPtr.reset(new ROIData(roiData));
+    tensorDesc = make_roi_tensor_desc(tensorDesc, roiPtr->roi);
+}
+
+Blob::Ptr Blob::CreateROIBlob(const ROI& roi) const {
+    Blob::Ptr roiBlob = std::shared_ptr<Blob>(this->clone());
```

Comment:
Nitpick: use `auto`, you already specified type on the right side of `=`.

Reply:
Is it possible to use `std::make_shared` here?

Reply:
I doubt so - the Blob::ctr is not what is needed here.

Reply:
Could you please clarify your point a bit? Do you mean there is implicit conversion and `Blob::Ptr` is not `std::shared_ptr<Blob>`?

Reply:
nope.
> make_shared
> Constructs an object of type T and wraps it in a std::shared_ptr using args as the parameter list for the constructor of T. The object is constructed as if by the expression ::new (pv) T(std::forward<Args>(args)...)

In contrast, here we clone an object of arbitrary Blob descendant.

Reply:
So, return type is `Blob::Ptr` and can be replaced with `auto` since it already mentioned on the right side of `=`?

Reply:
Somewhere here I said already that:
when the type name is not awkward, I prefer to see the real type and do not hide it behind auto.

Reply:
But `Blob::Ptr` is still awkward - you don't know which pointer type is here, right?

Reply:
It's not awkward because it's short.
The pointer type is `Blob::Ptr`. The object type behind the pointer does not matter here.

Reply:
Strange argument about length - `auto` is shorter

Reply:
>when the type name is not awkward, I prefer to see the real type and **do not hide it** behind auto.

Does it make sense?

Reply:
My point is even trying not to hide actual type here you still don't show since `Blob::Ptr` hides it. But it's probably subjective, if others are agree - fine :)
Thanks for discussion - resolving

Reply:
The type of pointer is `Blob::Ptr`; this is what matters for the reader. Actual type of the object is not hidden - it is just not available statically.